src/platform/site-wide/user-nav/containers/AutoSSO.jsx

```diff
@@ -1,23 +1,37 @@
+import React from 'react';
 import { connect } from 'react-redux';
 
 import { checkKeepAlive } from 'platform/user/authentication/actions';
 import {
   ssoeInbound,
   hasCheckedKeepAlive,
+  isAuthenticatedWithSSOe,
 } from 'platform/user/authentication/selectors';
+import { isProfileLoading } from 'platform/user/selectors';
 import { hasSession } from 'platform/user/profile/utilities';
 import { checkAutoSession } from 'platform/utilities/sso';
 import { removeLoginAttempted } from 'platform/utilities/sso/loginAttempted';
 
 function AutoSSO(props) {
-  const { useInboundSSOe, hasCalledKeepAlive } = props;
+  const {
+    useInboundSSOe,
+    hasCalledKeepAlive,
+    authenticatedWithSSOe,
+    profileLoading,
+  } = props;
 
   if (hasSession()) {
     removeLoginAttempted();
   }
 
-  if (useInboundSSOe && !hasCalledKeepAlive) {
-    checkAutoSession().then(() => {
+  if (
+    // avoid race condition where hasSession hasn't been set
+    !window.location.pathname.includes('auth/login/callback') &&
+    useInboundSSOe &&
+    !profileLoading &&
```

Comment:
will this check make inbound ssoe slower?

Reply:
It shouldn't. My [longer response](https://github.com/department-of-veterans-affairs/vets-website/pull/13481/files#r455183069) to Patrick's comments below gives more context, but `profileLoading` is tied to the `/v0/user` call, if it's required, and since the ssoe flag is part of that call, we should act until we know the state of that anyway